src/main/java/entities/CollaborativeTask.java

```diff
@@ -0,0 +1,217 @@
+     * @param oldTeammates - an Array List of Student Users who are no longer undergoing the process of being
+     *                       invited to join the Collaborative Task.
+     */
+    protected void removePendingTeammates(ArrayList<StudentUser> oldTeammates){
+        this.pendingTeammates.removeAll(oldTeammates);
+    }
+
+    /**
+     * @return an Array List of all Student Users who have declined invitations to join the Collaborative Task.
+     */
+    protected ArrayList<StudentUser> getDeclinedTeammates(){
+        return this.declinedTeammates;
+    }
+
+    /**
+     * To be used when invitations to join a Collaborative Task are declined.
+     * @param newTeammates - an Array List of Student Users who have declined invitations to join the Collaborative Task.
+     */
+    protected void addDeclinedTeammates(ArrayList<StudentUser> newTeammates){
+        this.declinedTeammates.addAll(newTeammates);
+    }
+
+    /**
+     * To be used when a Student User who has previously declined an invitation to join the Collaborative Task has
+     * accepted a reinvitation to join.
+     * @param oldTeammates - an Array List of Student Users who have accepted an invitation to join the Collaborative
+     *                       Task after previously declining such an invitation.
+     */
+    protected void removeDeclinedTeammates(ArrayList<StudentUser> oldTeammates){
+        this.declinedTeammates.removeAll(oldTeammates);
+    }
+
+    /**
+     * Set a Collaborative Task as recurring/not
+     * Set its new frequency if it is recurring
+     * @param recurring - whether the Collaborative Task is recurring
+     * @param frequency - the frequency at which the Collaborative Task occurs (if recurring)
+     */
+    protected void setRecurring(boolean recurring, String frequency) {
+        this.recurring = recurring;
+        if (recurring) {
+            this.frequency = frequency;
+        }
+        else{
+            this.frequency = "";
+        }
+    }
+
+    /**
+     * Set a new time block
+     * @param startTime - the start of the time block
+     * @param endTime - the end of the time block
+     */
+    public void setTimeBlock(LocalDateTime startTime, LocalDateTime endTime) {
+
+    }
+
+    /**
+     * Get the time block of an Event
+     * @return - the time block of the Event
+     *         - in array form: {startTime, endTime}
+     */
+    public LocalDateTime[] getTimeBlock() {
+        return new LocalDateTime[] {this.startTime, this.endTime};
+    }
+
+    /**
+     * Schedule a time block for the user
+     * @return - whether the time block has been successfully scheduled
+     */
+    public boolean scheduleTimeBlock() {
+        return true;
+    }
+
+    /**
+     * Remove a time block from the user's schedule
+     * @return - whether the time block has been successfully removed
+     */
+    public boolean removeTimeBlock() {
+        return true;
+    }
+
```

Comment:
I ended up removing the save, edit and delete abstract methods from Task after seeing that they are more fit for task use cases than task. That being said, I kept save in Task itself but made it not abstract (saves tasks to the task map).

Reply:
Deleted those methods from Collaborative Task.